src/frame.js

```diff
@@ -0,0 +1,127 @@
+class Frame {
+  constructor() {
+    this.roll1 = null;
+    this.roll2 = null;
+    this.roll3 = null;
+    this.frameScore = null;
+    this.strike = false;
+    this.spare = false;
+    this.frameFinished = false;
```

Comment:
There is information duplication in your class. The score, strike, spare and finished could be calculated on the fly based on the rolls. Calculating on the fly is better, because it avoids information getting out of sync between variables, and ultimately bugs are less likely.

Reply:
So basically, this class should only have rolls 1 to 3 as instance variables, everything else should be methods. here's one example of such a method:
```js
strike() {
  return this.roll1 === 10;
}
```

Reply:
Basically, it's almost like reversing the flow. Here you are receiving a roll, and setting all variables. The other way around you only set the roll, and then everything is calculated when you need it / request it. Potentially less computing to do, tot.